### PR TITLE
[Merged by Bors] - Migrate integration tests

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- Moved tests to operator repo
- Adapted to [ADR018](https://docs.stackable.tech/home/contributor/adr/ADR018-product_image_versioning.html)

## Review Checklist

- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by backstreetkiwi — APPROVED** [review]
successfully run: https://ci.stackable.tech/view/02%20Component%20Tests%20(flex)/job/Spark%20Operator%20Integration%20Tests%20(flex)/30/

File: rust/operator-binary/src/spark_controller.rs [hunk 1]
```diff
@@ -319,12 +321,9 @@ fn build_rolegroup_statefulset(
     rolegroup_ref: &RoleGroupRef<SparkCluster>,
     rolegroup_config: &HashMap<PropertyNameKind, BTreeMap<String, String>>,
 ) -> Result<StatefulSet, Error> {
-    let version = version(spark)?;
+    let version = image_version(spark)?;
```
Comment (sbernauer) [inline rust/operator-binary/src/spark_controller.rs:324]: Naming it `image_version` or `spark_image_version` would be great

File: rust/operator-binary/src/spark_controller.rs [hunk 2]
```diff
@@ -480,10 +479,24 @@ fn convert_map_to_string(map: &BTreeMap<String, String>, assignment: &str) -> St
...
 }
+
+/// Returns our semver representation for product config e.g. 3.1.1
+fn product_version(spark: &SparkCluster) -> Result<&str, Error> {
+    let image_version = image_version(spark)?;
+    image_version
+        .split('-')
+        .collect::<Vec<_>>()
+        .first()
+        .cloned()
```
Comment (sbernauer) [inline rust/operator-binary/src/spark_controller.rs:498]:
```suggestion
        .next()
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,10 +9,12 @@ All notable changes to this project will be documented in this file.
 - Reconciliation errors are now reported as Kubernetes events ([#289]).
 - Use cli argument `watch-namespace` / env var `WATCH_NAMESPACE` to specify
   a single namespace to watch ([#297]).
+- Moved tests from integration tests repo to operator repo ([#344]).
 
 ### Changed
 
-- `operator-rs` `0.10.0` -> `0.15.0` ([#289], [#291], [#297], [#326]).
+- `operator-rs` `0.10.0` -> `0.21.1` ([#289], [#291], [#297], [#326], [#344]).
```
Comment (sbernauer) [inline CHANGELOG.md:16]: Time for a release, isn't it? :joy:

**Review by sbernauer — APPROVED** [review]
LGTM % comments

File: rust/operator-binary/src/spark_controller.rs (hunk 1 above)
Comment (maltesander) [reply rust/operator-binary/src/spark_controller.rs]: done

File: rust/operator-binary/src/spark_controller.rs (hunk 2 above)
Comment (maltesander) [reply rust/operator-binary/src/spark_controller.rs]: done

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**